### ISSUE TITLE
IAM user and group names should allow underscore 

### ISSUE THREAD

**Issue by bandesz** [hashicorp/terraform#9679]
The https://github.com/hashicorp/terraform/pull/9584 changeset adds validation for the IAM user and group name, but the AWS official documentation is not correct and you can create users/groups with underscores in the name. This changeset will probably force some users to change names or not upgrade.

### Terraform Version
Terraform v0.7.8-dev (3500630243a167bf0b88b18bf27a221e2f7c6707)

### Affected Resource(s)
aws_iam_user
aws_iam_group

### Terraform Configuration Files
```hcl
resource "aws_iam_user" "some_user" {
    name = "some_user"
}
resource "aws_iam_group" "some_group" {
    name = "some_group"
}
```

### Expected Behavior
Names with underscores should be allowed. (Even the username "_" is allowed)

### Actual Behavior
Name validation fails.

### Steps to Reproduce
terraform apply

### References
- GH-9584

**Comment by bandesz** [comment]
I submitted a PR for this: https://github.com/hashicorp/terraform/pull/9684

**Comment by stack72** [comment]
Closed via #9684
